docs/DeveloperGuide.md

```diff
@@ -257,14 +263,58 @@ Command : `find n/ Harry`
   the user uses the keyword for multiple fields.
   * Cons: Might not be easy to find specific books, i.e. cannot streamline the search as well.
 
+### Sort feature
+
+#### Implementation 
+
+*bookmark* allows Users to sort books by a specified field.
+
+This feature is facilitated mainly by `ModelManager`, `SortCommandParser` and `SortCommand`.
+
+`ModelManager#sortFilteredBookList()` takes in a single parameter, a `comparator`, and applies the `comparator` on all elements
+of the internal observable book list. Book's are sorted according to the input comparator in ascending order. 
+
+`ModelManager#setSortingPreference()` takes in a single parameter, a `newSortingPreference`, and updates the sorting preference 
+in the `preferences.json` file. Thereafter, the specified sorting mechanism will apply to the user until the user decides to change 
+the sorting mechanism using the sort command. 
+
+Currently, the `sort` command supports sorting by name, genre, bookmark, and reading progress.
 
+#### Sorting the ObservableList
+The `SortCommandParser#parse()` parses the `sort` command input, and checks for input errors for which if found,
+would throw an error. Subsequently, `ComparatorGenerator#comparatorGenerator()` generates a comparator based on the 
+user's `inputPrefix`. `inputPrefix` and the resultant `comparator` are used to generate a new `SortCommand` object.
+When `SortCommand#execute()` is called, `inputPrefix` is passed to `ModelManager#setSortingPreference()`, where 
+updating of user sorting preferences in `preferences.json` occurs. `comparator` is passed to `ModelManager#sortFilteredBookList()`,
+where sorting of the internal observable list based on the `comparator` occurs.
+
+The activity diagram below illustrates the flow of execution when the user inputs a `sort` command.
+
+![SortActivityDiagram](images/SortActivityDiagram.png)
+
+Below is a sequence diagram that shows a scenario whereby the user decides to sort books by the name field.
+Command : `sort n/`
+
+![SortSequenceDiagram](images/SortSequenceDiagram.png)
+
+##### Aspect: Sorting visible observable list or internal observable list
+
+* **Alternative 1 (current choice):** Sorting internal observable list
+  * Pros: Allows user to set sorting preference. When the user exits and reopens the application, the books would remain
+  in the sorted order.
+  * Cons: Could be a drawback if the user only wants to sort for temporary view.
+  
+* **Alternative 2:** Sorting visible observable list
+  * Pros: Sorts the user's book list temporarily, which is useful if the user only wants the sorted view momentarily.
+  * Cons: Book list would always return to default view after subsequent commands which is be distracting.
```

Comment:
Typo here? the "which is be distracting" part

Reply:
Yup that's a typo, nice catch!